### ISSUE TITLE
Segmentation fault VS Code Server for WSL closed unexpectedly.

### ISSUE THREAD

**Issue by relston** [microsoft/vscode-remote-release#5059]
After updating to Windows 10 Insider Preview 21382.1  I am no longer able to connect to WSL. It errors with a `Segmentation fault` when trying to connect. 

Here are the logs 
```bash
[2021-05-15 23:43:10.542] Resolving wsl+debian, resolveAttempt: 1
[2021-05-15 23:43:10.614] Starting VS Code Server inside WSL (Debian)
[2021-05-15 23:43:10.614] Extension version: 0.56.2, Windows build: 21382. Multi distro support: available. WSL path support: enabled
[2021-05-15 23:43:10.614] No shell environment set or found for current distro.
[2021-05-15 23:43:10.803] Probing if server is already installed: C:\WINDOWS\System32\wsl.exe -d Debian -e sh -c "[ -d ~/.vscode-server/bin/054a9295330880ed74ceaedda236253b4f39a335 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
[2021-05-15 23:43:11.100] Probing result: found
[2021-05-15 23:43:11.100] Server install found in WSL
[2021-05-15 23:43:11.101] Launching C:\WINDOWS\System32\wsl.exe -d Debian sh -c '"$VSCODE_WSL_EXT_LOCATION/scripts/wslServer.sh" 054a9295330880ed74ceaedda236253b4f39a335 stable .vscode-server 0  '}
[2021-05-15 23:43:11.418] Setting up server environment: Looking for /home/relston/.vscode-server/server-env-setup. Not found.
[2021-05-15 23:43:11.418] WSL version: 4.19.128-microsoft-standard Debian
[2021-05-15 23:43:11.418] WSL2-shell-PID: 622
[2021-05-15 23:43:11.418] Starting server: /home/relston/.vscode-server/bin/054a9295330880ed74ceaedda236253b4f39a335/server.sh  --port=0 --use-host-proxy --without-browser-env-var --print-ip-address --enable-remote-auto-shutdown  
[2021-05-15 23:43:11.430] Segmentation fault
[2021-05-15 23:43:11.431] VS Code Server for WSL closed unexpectedly.
[2021-05-15 23:43:11.431] For help with startup problems, go to
[2021-05-15 23:43:11.431] https://code.visualstudio.com/docs/remote/troubleshooting#_wsl-tips
[2021-05-15 23:43:11.436] C:\WINDOWS\System32\wsl.exe -d Debian -e kill 622
[2021-05-15 23:43:11.666] WSL Daemon exited with code 0
```

I did find the [The server fails to start with a segmentation fault](https://code.visualstudio.com/docs/remote/troubleshooting#_the-server-fails-to-start-with-a-segmentation-fault) section and I am able to get a core dump, but there is nothing in this article that suggests what to do with that dump or how it would be helpful. 

The dump is > 45mbs and doesn't seem like something I can upload here. If yall think it's going to be useful I can find another way to share it. 

How can I get VS Code to work with WSL again?

**Comment by bamurtaugh** [comment]
@aeschli Would the core dump be useful? 

I also wonder if there's something in the recent Windows update that could be causing connection issues - https://github.com/microsoft/vscode-remote-release/issues/4783, https://github.com/microsoft/vscode-remote-release/issues/5066, https://github.com/microsoft/vscode-remote-release/issues/4932 seem potentially related

**Comment by aeschli** [comment]
I'm trying to reproduce by updating my Windows insiders build...

To be sure we don't have a corrupt install, try this:
- close VSCode
- `rm -r /home/relston/.vscode-server/bin`

**Comment by aeschli** [comment]
Windows insiders build 21390 works for me.

If you still see the seg fault, it would be great if you can give me access to the file.

**Comment by relston** [comment]
Thanks for getting back to me! Really appreciate it. Sorry for not responding sooner. 

TL;DR: `rm -r /home/relston/.vscode-server/bin` did fix the issue for me. Maybe we can escalate that solution in case others run into this?

I don't know if it's helpful but I noticed some peculiar behavior while facing this issue.

For example, opening a project from my WSL home directory (`/home/relston/Projects/whatever`) and VSCode would Segment Fault. But open the same project folder from a WSL windows network path (`\\wsl.localhost\Debian\home\relston\Projects\whatever`) and VSCode was able to open a Linux terminal. Weird, but it made for a good workaround for the time being. The problem with that was that none of the git integration worked so it was pretty annoying but manageable. 

Anyway, thanks for the help!

**Comment by aeschli** [comment]
@relston Opening `/home/relston/Projects/whatever` in a WSL window and opening `\\wsl.localhost\Debian\home\relston\Projects\whatever` in a normal (non-remote) window are very different.
I the first case, we run the VSCode back-end (the vscode server) in the WSL distro, on Linux, but in the second case it's running on the window side. For the first, all extensions and tools (including git) run in Linux, which is what you want when you develop in WSL.

I'm closing the issue. Please reopen if you get a seg fault again.
